### DISCUSSION TITLE
Can Bitwarden access vaultwarden through authentik or security proxy?

### DISCUSSION THREAD

**Discussion by SpiderUnderUrBed** [dani-garcia/vaultwarden#4611] [Q&A] [answered]
The title is my question, I am planning to set up authentik to secure my homelab applications, I am using tailscale to access them and I shared my exit node with some of my friends, so for security reasons its important I sort out permissions and what services they can access, but I am worried that if i set up authentik to be the only way to access vaultwarden (I think making another way to access it directly that only I can use is kinda infeasable/I dont know how to do it) 

Then bitwarden wont be able to access the vault. How should I set this up? 
(This is unrelated to SSO but OIDC or some key given in the vaults url server header to bypass authentiks protection could help)

**Comment by crowley69** [comment, marked as answer]
I hope this proves useful.

My Vaultwarden is in a docker container behind nginx, and it is setup in authentik as a Forward auth proxy provider.
I am able to protect Vaultwarden web vault as well as the admin page while being able to use the Bitwarden app/extension normally by allowing the API to bypass authentik.

In Authentik > Provider > Advanced protocol settings > Unauthenticated Paths
I simply allowed the API by adding `^/api/.*` to the Unauthenticated Paths, I also allowed a few other paths as well:

```
^/api/.*
^/wl/.*
^/identity/.*
^/images/.*
```
With this the Bitwarden app will bypass authentik flow, but you won't be able to use the web vault login, admin page or sends without authentication. 


If your only goal is to protect the admin page behind authentik, the following regular expressions are all you need:

```
^/$
^/.*\.js
^/#/.*
^/#/login
^/#/2fa
^/#/vault
^/api/.*
^/images/.*
^/identity/.*
^/app/.*
^/locales/.*
```

Results may vary depending on your setup.

**Reply by MrDuartePT** [reply]
I add that option to the Unauthenticated Paths but the bitwarden app dosent let me login. And sync on my phone gives an error.

**Reply by dedsxc** [reply]
The regex provided didnt work for me. Updated to those regex instead to make it works: 

```bash
^/api(/.*)?$
^/wl(/.*)?$
^/identity(/.*)?$
^/images(/.*)?$
```

**Comment by BlackDex** [comment]
To make it a bit clear. It will not prevent an attacker if they really want to access your environment.
Since the API endpoints are still accessible, it will only prevent the web-interface from being directly available.

**Reply by masterwishx** [reply]
Do you mean is  better without Authentik?
I mean without Authentik the API also avaliable?

**Reply by luc-hart** [reply]
Depending on your risk appetite, the following solutions would be lower risk and allows you to not set any unauthenticated paths (better security-wise):

1. Use a VPN on your server and your devices (tailscale or wg-easy for example)
OR
2. Whitelist your home network IP in your reverse proxy settings and use vaultwarden in read-only mode when not at home. This way everything will work fine, just the synching of passwords will only happen when you're at home.

**Comment by Guruleenyc** [comment]
What about using OIDC with Authentik?

I configured these Vaultwarden docker variables, but I am still not getting the 'login with sso' option at the login screen.

SSO_ENABLED: "true"
  SSO_OVERRIDE_PASSWORD_LOGIN: "false" # Set to "true" to disable password login entirely
  SSO_TYPE: "openidconnect"
  SSO_OIDC_AUTO_DISCOVER_URL: "https://your.authentik.domain.com/application/o/vaultwarden/.well-known/openid-configuration" # The URL copied from Authentik
  SSO_OIDC_CLIENT_ID: "YOUR_CLIENT_ID_FROM_AUTHENTIK" # The Client ID copied from Authentik
  SSO_OIDC_CLIENT_SECRET: "YOUR_CLIENT_SECRET_FROM_AUTHENTIK" # The Client Secret copied from Authentik
  SSO_OIDC_REDIRECT_URL: "https://your.vaultwarden.domain.com/sso/oidc-signin" # Your Vaultwarden URL + /sso/oidc-signin
  SSO_OIDC_SCOPES: "openid profile email" # Matches the scopes you set in Authentik
  SSO_ROLES_ENABLED: "false" # Set to "true" if you want to map Authentik groups to Vaultwarden roles (admin/user)
  SSO_ROLES_TOKEN_PATH: "/groups" # Path in the OIDC token for group information (if SSO_ROLES_ENABLED is true)
  SSO_ROLES_ADMIN_GROUP: "vaultwarden_admins" # Authentik group name for admin role (if SSO_ROLES_ENABLED is true)
  SSO_ROLES_USER_GROUP: "vaultwarden_users" # Authentik group name for user role (if SSO_ROLES_ENABLED is true)

**Reply by violoncelloCH** [reply]
the feature is not yet released to stable according to https://github.com/dani-garcia/vaultwarden/pull/3899#pullrequestreview-3102172781 let's wait for the 2-4 weeks and see :)
or otherwise use the testing container image...

**Reply by violoncelloCH** [reply]
also, your `SSO_OIDC_AUTO_DISCOVER_URL` should not include the `/.well-known/openid-configuration` part according to the documentation: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-SSO-support-using-OpenId-Connect
